### PR TITLE
Added checking log_loss & roc_auc metrics in RF & kNN tests

### PR DESCRIPTION
**Pull request by OnlyDeniko**

### CODE REVIEW

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 1]
```diff
@@ -75,15 +77,15 @@
 ]
 
 
-def check_classifier_class_weight_iris(weight):
+def check_accuracy_classifier_class_weight_iris(weight):
     for _ in range(N_TRIES):
```
Comment (Alexsandruss) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:81]: Why do not check results stability run by run?
Applicable for other checks too

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 2]
```diff
@@ -75,15 +77,15 @@
...
         x_train, x_test, y_train, y_test = \
-            train_test_split(IRIS.data, IRIS.target, 
-                             test_size=0.3, random_state=31)
+            train_test_split(IRIS.data, IRIS.target,
+                             test_size=0.33, random_state=31)
 
-        scikit_model = ScikitRandomForestClassifier(class_weight=weight, 
+        scikit_model = ScikitRandomForestClassifier(class_weight=weight,
```
Comment (Alexsandruss) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:86]: Is algorithm parameters absence needed? It's better to specify to avoid problems when their default values are changed
Applicable for other checks too

File: daal4py/sklearn/ensemble/tests/test_decision_forest.py [hunk 3]
```diff
@@ -93,28 +95,84 @@ def check_classifier_class_weight_iris(weight):
...
+                                                   random_state=777)
+
+        scikit_predict_proba = scikit_model.fit(x_train, y_train).predict_proba(x_test)
+        daal4py_predict_proba = daal4py_model.fit(x_train, y_train).predict_proba(x_test)
+
+        scikit_roc_auc = roc_auc_score(y_test, scikit_predict_proba, multi_class='ovr')
+        daal4py_roc_auc = roc_auc_score(y_test, daal4py_predict_proba, multi_class='ovr')
+
+        ratio = daal4py_roc_auc / scikit_roc_auc
+        assert ratio >= CHECK_ROC_AUC_RATIO, \
```
Comment (Alexsandruss) [inline daal4py/sklearn/ensemble/tests/test_decision_forest.py:153]: Minor: warning or debug message if sklearn is worse than daal4py will be good

File: daal4py/sklearn/neighbors/tests/test_kneighbors.py [hunk 4]
```diff
@@ -18,19 +18,23 @@
...
 ALGORITHMS = ['brute', 'kd_tree', 'auto']
 WEIGHTS = ['uniform', 'distance']
-KS = [1, 3, 7, 25]
+KS = [1, 3, 7, 15, 31]
 N_TRIES = 10
-CHECK_RATIO_KNN = 0.85
-
+ACCURACY_RATIO = 0.85
+LOG_LOSS_RATIO = 1.006
+ROC_AUC_RATIO = 0.987
```
Comment (SmirnovEgorRu) [inline daal4py/sklearn/neighbors/tests/test_kneighbors.py:36]: Why not 1.0 here?

0.85 accuracy ration is too high

**Review by owerbat — APPROVED** [review]
